packages/mui-utils/src/integerPropType.d.ts

```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+declare function integerPropType(
```

Comment:
How is this related to the rest of this PR?

Reply:
I have pushed a commit without the file to illustrate the error. 

<img width="944" alt="Screenshot 2023-09-10 at 02 49 25" src="https://github.com/mui/material-ui/assets/3165635/f61af00e-4e9e-45f1-b978-d612dc729d51">

https://app.circleci.com/pipelines/github/mui/material-ui/107998/workflows/b65b0195-f80f-4407-9c86-0a1cfa39d6e2/jobs/578294